iOS/Flows/Conversation/Emotions/EmotionsViewController.swift

```diff
@@ -85,35 +76,94 @@ class EmotionsViewController: DiffableCollectionViewController<EmotionCategory,
         }
     }
     
-    override func layoutCollectionView(_ collectionView: UICollectionView) {
-        collectionView.expandToSuperviewWidth()
-        collectionView.height = self.view.height * 0.6
-        collectionView.pin(.bottom)
+    /// The currently running task that is loading conversations.
+    private var loadTask: Task<Void, Never>?
+    
+    private func removeLastEmotion() {
+        self.loadTask?.cancel()
+        
+        self.selectedEmotions.removeFirst()
+        
+        self.loadTask = Task { [weak self] in
+            guard let `self` = self else { return }
+            
+            var snapshot = self.dataSource.snapshot()
+            
+            var contentItems: [EmotionsCollectionViewDataSource.ItemType] = self.selectedEmotions.compactMap({ emotion in
+                return .emotion(EmotionContentModel(emotion: emotion))
+            })
+            
+            if contentItems.isEmpty {
+                contentItems = [.emotion(EmotionContentModel(emotion: nil))]
+            }
+            
+            snapshot.setItems(contentItems, in: .content)
+            
+            let categoryItems: [EmotionsCollectionViewDataSource.ItemType] = EmotionCategory.allCases.compactMap { category in
+                let model = EmotionCategoryModel(category: category, selectedEmotions: self.selectedEmotions)
+                return .category(model)
+            }
+            
+            snapshot.setItems(categoryItems, in: .categories)
+            await self.dataSource.apply(snapshot)
+        }
+    }
+    
+    private func handleSelected(emotion: Emotion) {
+        if self.selectedEmotions.contains(emotion) {
+            self.selectedEmotions.remove(object: emotion)
+        } else {
+            self.selectedEmotions.insert(emotion, at: 0)
```

Comment:
Here is where we insert at the beginning to ensure the most recent selection is visible.